cpp/src/table/row_operators.cu

```diff
@@ -28,6 +30,52 @@ namespace experimental {
 
 namespace {
 
+/**
+ * @brief Applies the offsets of struct columns onto their children.
+ *
+ * Given a table, this replaces any struct columns with similar struct columns that have their
+ * offsets applied to their children. Structs that are children of list columns are not affected.
+ *
+ */
+table_view pushdown_struct_offsets(table_view table)
+{
+  std::function<std::vector<column_view>(column_view const&)> slice_children =
+    [&](column_view const& c) -> std::vector<column_view> {
+    if (c.type().id() == type_id::STRUCT) {
+      std::vector<column_view> sliced_children;
+      sliced_children.reserve(c.num_children());
+      auto struct_col = structs_column_view(c);
+      for (size_type i = 0; i < struct_col.num_children(); ++i) {
+        auto sliced = struct_col.get_sliced_child(i);
+        // The reason we don't just copy the logic of structs_column_view::get_sliced_child() is
+        // because its logic might change in the future. The following logic is merely to replace
+        // the children of an existing column_view without changing anything else.
+        sliced_children.emplace_back(sliced.type(),
+                                     sliced.size(),
+                                     sliced.head<uint8_t>(),
+                                     sliced.null_mask(),
+                                     sliced.null_count(),
+                                     sliced.offset(),
+                                     slice_children(sliced));
+      }
```

Comment:
`transform` over `struct_col.child_begin/child_end`?

Reply:
Need the index for `get_sliced_child`